lib/streamlit/elements/write.py

```diff
@@ -204,6 +205,9 @@ def flush_buffer():
             elif isinstance(arg, Exception):
                 flush_buffer()
                 self.dg.exception(arg)
+            elif isinstance(arg, QueryParams):
+                flush_buffer()
+                self.dg.json(arg)
```

Comment:
this should probably be added to the `elif` below on L241 that currently checks if `isinstance(arg, (dict, list, SessionStateProxy, UserInfoProxy))` rather than having a new `elif` case